src/utils/managing.officer.corporate.ts

```diff
@@ -44,20 +45,31 @@ export const getManagingOfficerCorporateById = (req: Request, res: Response, nex
   try {
     logger.debugRequest(req, `${req.method} BY ID ${req.route.path}`);
 
+    const appData = getApplicationData(req.session);
+    const inUpdateJourney = EntityNumberKey in appData && appData[EntityNumberKey] !== undefined;
+
     const id = req.params[ID];
-    const data = getFromApplicationData(req, ManagingOfficerCorporateKey, id, true);
+    const officerData = getFromApplicationData(req, ManagingOfficerCorporateKey, id, true);
 
-    const principalAddress = (data) ? mapDataObjectToFields(data[PrincipalAddressKey], PrincipalAddressKeys, AddressKeys) : {};
-    const serviceAddress = (data) ? mapDataObjectToFields(data[ServiceAddressKey], ServiceAddressKeys, AddressKeys) : {};
+    const principalAddress = officerData ? mapDataObjectToFields(officerData[PrincipalAddressKey], PrincipalAddressKeys, AddressKeys) : {};
+    const serviceAddress = officerData ? mapDataObjectToFields(officerData[ServiceAddressKey], ServiceAddressKeys, AddressKeys) : {};
+    const startDate = officerData ? mapDataObjectToFields(officerData[StartDateKey], StartDateKeys, InputDateKeys) : {};
```

Comment:
Should the start date also only be for the update journey?

Reply:
I've wrapped this with a check to see if the MO is one that's just been added (i.e. has no ch_reference). This is the same now as the Individual Managing Officer.

Reply:
Resolving as approval given.